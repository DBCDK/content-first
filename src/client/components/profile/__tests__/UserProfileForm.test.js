/* eslint-disable no-undefined */
import React from 'react';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';

import UserProfileForm from '../UserProfileForm.component';
jest.mock('../../general/Spinner.component', () => 'spinner');

describe('UserProfileForm', () => {
  describe('render function', () => {
    it('renders an user profile form', () => {
      const tree = renderer.create(<UserProfileForm />).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('It adds validation error', done => {
      const tree = mount(<UserProfileForm />);
      const form = tree.find('form').first();
      form.simulate('submit');
      tree.update();
      expect(tree).toMatchSnapshot();
      tree.setProps({acceptedTerms: true});
      form.simulate('submit');
      tree.update();
      expect(tree).toMatchSnapshot();
      done();
    });
    it('It submits form', done => {
      const updateProfile = jest.fn();

      const tree = mount(
        <UserProfileForm
          acceptedTerms={true}
          name="some_name"
          imageId="some_image_id"
          updateProfile={updateProfile}
        />
      );
      tree
        .find('form')
        .first()
        .simulate('submit');
      expect(updateProfile.mock.calls[0][0]).toEqual({
        acceptedTerms: true,
        image: 'some_image_id',
        name: 'some_name'
      });
      done();
    });
  });
});
