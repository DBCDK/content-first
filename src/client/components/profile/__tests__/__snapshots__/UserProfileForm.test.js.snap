// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfileForm render function It adds validation error for age 1`] = `
<UserProfileForm>
  <form
    className="profile-form needs-validation"
    onSubmit={[Function]}
    style={
      Object {
        "maxWidth": "400px",
      }
    }
  >
    <div
      className="input-group mb2   has-feedback"
    >
      <input
        className="form-control mb3   has-feedback"
        data-cy="user-form-name"
        name="name"
        onChange={[Function]}
        placeholder="Vælg brugernavn"
        type="text"
        value=""
      />
      <span
        aria-hidden="true"
        className="form-control-feedback "
      >
        <i
          className="material-icons"
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           
        </i>
      </span>
    </div>
    <p
      className="mb6"
    >
      Du er logget på via 
    </p>
    <label
      className="checkbox"
      data-cy="user-form-acceptedAge"
      htmlFor="acceptedAge"
    >
      <input
        checked={false}
        className="checkbox"
        id="acceptedAge"
        name="acceptedAge"
        onChange={[Function]}
        type="checkbox"
      />
      <span />
       Jeg er over 13 år
    </label>
    <label
      className="checkbox"
      data-cy="user-form-acceptedTerms"
      htmlFor="acceptedTerms"
    >
      <input
        checked={false}
        className="checkbox"
        id="acceptedTerms"
        name="acceptedTerms"
        onChange={[Function]}
        type="checkbox"
      />
      <span />
       Jeg har læst og accepteret
       
      <a
        href="#terms"
      >
        reglerne for anvendelse af Læsekompasset
      </a>
    </label>
    <div
      className="error mb2"
    >
      Dit brugernavn skal være minimum 4 karakterer langt
    </div>
    <button
      className="btn btn-success btn-block"
      data-cy="user-form-submit"
    >
      Opret profil
       
    </button>
  </form>
</UserProfileForm>
`;

exports[`UserProfileForm render function It adds validation error for age 2`] = `
<UserProfileForm
  acceptedAge={true}
>
  <form
    className="profile-form needs-validation"
    onSubmit={[Function]}
    style={
      Object {
        "maxWidth": "400px",
      }
    }
  >
    <div
      className="input-group mb2   has-feedback"
    >
      <input
        className="form-control mb3   has-feedback"
        data-cy="user-form-name"
        name="name"
        onChange={[Function]}
        placeholder="Vælg brugernavn"
        type="text"
        value=""
      />
      <span
        aria-hidden="true"
        className="form-control-feedback "
      >
        <i
          className="material-icons"
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           
        </i>
      </span>
    </div>
    <p
      className="mb6"
    >
      Du er logget på via 
    </p>
    <label
      className="checkbox"
      data-cy="user-form-acceptedTerms"
      htmlFor="acceptedTerms"
    >
      <input
        checked={false}
        className="checkbox"
        id="acceptedTerms"
        name="acceptedTerms"
        onChange={[Function]}
        type="checkbox"
      />
      <span />
       Jeg har læst og accepteret
       
      <a
        href="#terms"
      >
        reglerne for anvendelse af Læsekompasset
      </a>
    </label>
    <div
      className="error mb2"
    >
      Dit brugernavn skal være minimum 4 karakterer langt
    </div>
    <button
      className="btn btn-success btn-block"
      data-cy="user-form-submit"
    >
      Opret profil
       
    </button>
  </form>
</UserProfileForm>
`;

exports[`UserProfileForm render function It adds validation error for terms 1`] = `
<UserProfileForm>
  <form
    className="profile-form needs-validation"
    onSubmit={[Function]}
    style={
      Object {
        "maxWidth": "400px",
      }
    }
  >
    <div
      className="input-group mb2   has-feedback"
    >
      <input
        className="form-control mb3   has-feedback"
        data-cy="user-form-name"
        name="name"
        onChange={[Function]}
        placeholder="Vælg brugernavn"
        type="text"
        value=""
      />
      <span
        aria-hidden="true"
        className="form-control-feedback "
      >
        <i
          className="material-icons"
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           
        </i>
      </span>
    </div>
    <p
      className="mb6"
    >
      Du er logget på via 
    </p>
    <label
      className="checkbox"
      data-cy="user-form-acceptedAge"
      htmlFor="acceptedAge"
    >
      <input
        checked={false}
        className="checkbox"
        id="acceptedAge"
        name="acceptedAge"
        onChange={[Function]}
        type="checkbox"
      />
      <span />
       Jeg er over 13 år
    </label>
    <label
      className="checkbox"
      data-cy="user-form-acceptedTerms"
      htmlFor="acceptedTerms"
    >
      <input
        checked={false}
        className="checkbox"
        id="acceptedTerms"
        name="acceptedTerms"
        onChange={[Function]}
        type="checkbox"
      />
      <span />
       Jeg har læst og accepteret
       
      <a
        href="#terms"
      >
        reglerne for anvendelse af Læsekompasset
      </a>
    </label>
    <div
      className="error mb2"
    >
      Dit brugernavn skal være minimum 4 karakterer langt
    </div>
    <button
      className="btn btn-success btn-block"
      data-cy="user-form-submit"
    >
      Opret profil
       
    </button>
  </form>
</UserProfileForm>
`;

exports[`UserProfileForm render function It adds validation error for terms 2`] = `
<UserProfileForm
  acceptedTerms={true}
>
  <form
    className="profile-form needs-validation"
    onSubmit={[Function]}
    style={
      Object {
        "maxWidth": "400px",
      }
    }
  >
    <div
      className="input-group mb2   has-feedback"
    >
      <input
        className="form-control mb3   has-feedback"
        data-cy="user-form-name"
        name="name"
        onChange={[Function]}
        placeholder="Vælg brugernavn"
        type="text"
        value=""
      />
      <span
        aria-hidden="true"
        className="form-control-feedback "
      >
        <i
          className="material-icons"
          style={
            Object {
              "fontSize": 18,
            }
          }
        >
           
        </i>
      </span>
    </div>
    <p
      className="mb6"
    >
      Du er logget på via 
    </p>
    <label
      className="checkbox"
      data-cy="user-form-acceptedAge"
      htmlFor="acceptedAge"
    >
      <input
        checked={false}
        className="checkbox"
        id="acceptedAge"
        name="acceptedAge"
        onChange={[Function]}
        type="checkbox"
      />
      <span />
       Jeg er over 13 år
    </label>
    <div
      className="error mb2"
    >
      Dit brugernavn skal være minimum 4 karakterer langt
    </div>
    <button
      className="btn btn-success btn-block"
      data-cy="user-form-submit"
    >
      Opret profil
       
    </button>
  </form>
</UserProfileForm>
`;

exports[`UserProfileForm render function renders an user profile form 1`] = `
<form
  className="profile-form needs-validation"
  onSubmit={[Function]}
  style={
    Object {
      "maxWidth": "400px",
    }
  }
>
  <div
    className="input-group mb2   has-feedback"
  >
    <input
      className="form-control mb3   has-feedback"
      data-cy="user-form-name"
      name="name"
      onChange={[Function]}
      placeholder="Vælg brugernavn"
      type="text"
      value=""
    />
    <span
      aria-hidden="true"
      className="form-control-feedback "
    >
      <i
        className="material-icons"
        style={
          Object {
            "fontSize": 18,
          }
        }
      >
         
      </i>
    </span>
  </div>
  <p
    className="mb6"
  >
    Du er logget på via 
  </p>
  <label
    className="checkbox"
    data-cy="user-form-acceptedAge"
    htmlFor="acceptedAge"
  >
    <input
      checked={false}
      className="checkbox"
      disabled={undefined}
      id="acceptedAge"
      name="acceptedAge"
      onChange={[Function]}
      type="checkbox"
    />
    <span />
     Jeg er over 13 år
  </label>
  <label
    className="checkbox"
    data-cy="user-form-acceptedTerms"
    htmlFor="acceptedTerms"
  >
    <input
      checked={false}
      className="checkbox"
      disabled={undefined}
      id="acceptedTerms"
      name="acceptedTerms"
      onChange={[Function]}
      type="checkbox"
    />
    <span />
     Jeg har læst og accepteret
     
    <a
      href="#terms"
    >
      reglerne for anvendelse af Læsekompasset
    </a>
  </label>
  <button
    className="btn btn-success btn-block"
    data-cy="user-form-submit"
    disabled={undefined}
  >
    Opret profil
     
    
  </button>
</form>
`;
